osu.Game/Overlays/SettingsPanel.cs

```diff
@@ -287,7 +287,10 @@ public string SearchTerm
                 set
                 {
                     SearchContainer.SearchTerm = value;
-                    InvalidateScrollPosition();
+
+                    // Schedule required as search term takes a frame to update.
+                    // Without this sections may not be in the correct state to ascertain CurrentSection.
+                    Schedule(InvalidateScrollPosition);
```

Comment:
`InvalidateScrollPosition` already schedules invalidating its properties internally ([and that schedule is seemingly done for the same purpose](https://github.com/ppy/osu/commit/af0c7ed108387b4e77ee8bf4f902dafbd2f12551)), so this is adding another layer of scheduling for things to work...

On further investigation, scheduling should never be required in theory, since `SearchContainer` is a child of `SectionsContainer`, and `SectionsContainer` updates the current section in `UpdateAfterChildren`.

The reason why scheduling is required is because [`SearchContainer` is faded from zero once content is loaded](https://github.com/ppy/osu/blob/316c0845ec8f56c92d36ea702038b3ec545e24e1/osu.Game/Overlays/SettingsPanel.cs#L212), causing the search container to be not present on first frame and not update the sections filter state, and leading to `SectionsContainer` revalidating position while the sections are not filtered.

One ideal solution I can suggest which removes scheduling from both here and inside the method is exposing an event from `SearchContainer` that's invoked once filtering is updated, and hook that to `InvalidateScrollPosition`.

Or alternatively, either make `SearchContainer` always present or expose its scheduler and schedule the scroll position invalidation to it, to ensure the sections are filtered properly by then.

I would personally go with a minor framework-side change for correctness rather than rely on more hacks.

Reply:
Making `SearchContainer` always present is the most acceptable solution for me out of all mentioned. Everything else is adding complexity.

Reply:
I personally find using `AlwaysPresent` here is just a workaround for assuming `SearchContainer` will always be updated per-frame, rather than not invalidating until the filtering has actually been done.

Reply:
the filtering logic should always work regardless of presence of the search container. consumers should not be need to subscribe to any events, or to schedule operations to special schedulers.

Reply:
I've added an event via https://github.com/ppy/osu-framework/pull/5116, I think this is the most amicable path forward.